apollo-federation/src/sources/connect/json_selection/apply_to.rs

```diff
@@ -341,28 +371,72 @@ impl ApplyTo for PathSelection {
     }
 }
 
+impl ApplyTo for LitExpr {
+    fn apply_to_path(
+        &self,
+        data: &JSON,
+        vars: &VarsWithPathsMap,
+        input_path: &InputPath<JSON>,
+        errors: &mut IndexSet<ApplyToError>,
+    ) -> Option<JSON> {
+        match self {
+            Self::String(s) => Some(JSON::String(s.clone().into())),
+            Self::Number(n) => Some(JSON::Number(n.clone())),
+            Self::Bool(b) => Some(JSON::Bool(*b)),
+            Self::Null => Some(JSON::Null),
```

Comment:
Is there a reason that we want to have a way to express null as a literal?

Reply:
Yes, `null` is already a JSON literal and quite useful for expressing intentional absence of objects, depending on your conventions.